airbyte-integrations/connectors/source-instatus/source_instatus/instatus.yaml

```diff
@@ -0,0 +1,268 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: [ ]
+
+  user_field_selector:
+    extractor:
+      field_pointer: [ "user" ]
+
+  requester:
+    url_base: "https://api.instatus.com/"
+    http_method: "GET"
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['api_key'] }}"
+
+  api_error_handler_requester:
+    $ref: "*ref(definitions.requester)"
+    error_handler:
+      type: CompositeErrorHandler
+      error_handlers:
+        - type: DefaultErrorHandler
+          response_filters:
+            # 401 This user is not an owner of the status page
+            - http_codes: [ 401 ]
+              action: IGNORE
+        - type: DefaultErrorHandler
+          response_filters:
+              - http_codes: [ 429 ]
+                action: RETRY
+          backoff_strategies:
+            - type: WaitTimeFromHeader
+              header: "Retry-After"
+
+  public_data_requester:
+    $ref: "*ref(definitions.api_error_handler_requester)"
+    url_base: "https://{{ stream_slice.url }}.instatus.com/"
+
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+
+  user_field_retriever:
+    record_selector:
+      $ref: "*ref(definitions.user_field_selector)"
+
+  schema_loader:
+    type: JsonSchema
+    file_path: "./source_instatus/schemas/{{ options['name'] }}.json"
+
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+
+  status_pages_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "status_pages"
+      primary_key: "id"
+      path: "v2/pages"
+
+  user_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "user"
+      primary_key: "id"
+      path: "v1/user"
+
+  page_stream_slicer:
+    type: SubstreamSlicer
+    parent_stream_configs:
+      - stream: "*ref(definitions.status_pages_stream)"
+        parent_key: "id"
+        stream_slice_field: "page_id"
+
+  page_components_stream:
+    schema_loader:
```

Comment:
instead of defining this in every stream, if you were to reference the schema loader in the base stream, you should be able to avoid defining it in every stream.

```
base_stream:
  retriever:
    $ref: "*ref(definitions.retriever)"
  schema_loader:
    $ref: "*ref(definitions.schema_loader)"
```

Reply:
thanks, done